### ISSUE TITLE
[Delta] Register operationMetrics for table history

### ISSUE THREAD

**Issue by osopardo1** [Qbeast-io/qbeast-spark#137]
## What went wrong?
The underlying format Delta Lake is capable of registering metrics of the different operations (WRITE, UPDATE, DELETE)... When writing with Qbeast Format, since we change the pipeline, we missed adding this information.

These metrics are useful to monitor the performance and analyze the behavior of the operations allowed. Here you can find the documentation: 
- Schema https://docs.databricks.com/delta/delta-utility.html#history-schema 
- Metrics https://docs.databricks.com/delta/delta-utility.html#operation-metrics-keys

## How to reproduce?

Different steps about how to reproduce the problem.

### 1. Code that triggered the bug, or steps to reproduce:

You can create or load test data into a dataFrame and write it using both Delta and Qbeast.

```scala
    val tmpDir = "/tmp/dir"
    val data = spark.read
    .format("csv")
    .option("header", "true")
    .option("inferSchema", "true")
    .load("src/test/resources/ecommerce100K_2019_Oct.csv")

    data.write.format("qbeast").option("columnsToIndex", "user_id,price").save(tmpDir + "/qbeast")
    data.write.format("delta").save(tmpDir + "/delta")

```

When retrieving the history of a Qbeast Table, the output shows some empty fields:
```scala
spark.sql(s"DESCRIBE HISTORY '$tmpDir/qbeast'").select("operationMetrics").show(false)
+----------------+
|operationMetrics|
+----------------+
|{}              |
+----------------+

```

But when analyzing a Delta Table:

```scala
spark.sql(s"DESCRIBE HISTORY '$tmpDir/delta'").select("operationMetrics").show(false)
+------------------------------------------------------------------+
|operationMetrics                                                  |
+------------------------------------------------------------------+
|{numFiles -> 8, numOutputRows -> 99986, numOutputBytes -> 4303479}|
+------------------------------------------------------------------+

```
### 2. Branch and commit id:
`main` 

### 3. Spark version:
On the spark shell run `spark.version`.

3.2.2

### 4. Hadoop version:
On the spark shell run `org.apache.hadoop.util.VersionInfo.getVersion()`.

3.2.1

### 5. How are you running Spark?
Are you running Spark inside a container? Are you launching the app on a remote K8s cluster? Or are you just running the tests in a local computer?

Local spark
### 6. Stack trace:
Trace of the log/error messages.

NA

**Comment by osopardo1** [comment]
Here are my current thoughts:

- We are missing `statsTrackers`( on the `BlockWriter` and `SparkDeltaDataWriter`. We need to think about how to include them.
- This `statsTrackers` would also help with issue #4 
- It can be possible to refactor the way we write the files by using the class from Spark `FileFormatWriter`, which does all the logic for managing those stats. An alternative is to implement ourselves when to update those stats (I saw that those stats are collected mainly when a new file is created and when rows are written to the file).

**Comment by osopardo1** [comment]
Fixed in #139
